File: poh21/3.rb
#!/usr/bin/ruby
X,Y,Z=gets.split.map &:to_i
r=Z
x=0
while x*X<=Z
  y=0
  while x*X+y*Y<=Z
    r=[r,x+y+Z-(x*X+y*Y)].min
    y+=1
  end
  x+=1
end
p r

File: poh20/java/3.java
//usr/bin/env java $0 $@;exit
import java.util.*;
interface Main{static void main(String[]a){Scanner x=new Scanner(System.in);System.out.println(Arrays.asList(x.nextLine().split(" ")).stream().mapToInt(e->Integer.parseInt(e)).sum());}}
